slack_bolt/app/app.py

```diff
@@ -58,21 +58,11 @@ def __init__(
         client: Optional[WebClient] = None,
         # for multi-workspace apps
         installation_store: Optional[InstallationStore] = None,
-        oauth_state_store: Optional[OAuthStateStore] = None,
-        oauth_state_cookie_name: str = OAuthStateUtils.default_cookie_name,
-        oauth_state_expiration_seconds: int = OAuthStateUtils.default_expiration_seconds,
         # for the OAuth flow
+        oauth_state_store: Optional[OAuthStateStore] = None,
+        oauth_settings: Optional[OAuthSettings] = None,
         oauth_flow: Optional[OAuthFlow] = None,
```

Comment:
Everything looks good, but there are two points I'm not clear on conceptually:

1. With the introduction of `oauth_settings`, what is the relationship between that and `oauth_flow`? I notice that `oauth_flow` takes `settings` as an optional parameter of `OAuthSettings`, as well, so are there multiple ways for the dev to accomplish the same end?
2. Why is `oauth_state_store` available as an argument outside of `oauth_settings`?

Reply:
@misscoded 

>1 With the introduction of oauth_settings, what is the relationship between that and oauth_flow? I notice that oauth_flow takes settings as an optional parameter of OAuthSettings, as well, so are there multiple ways for the dev to accomplish the same end or do these things do fundamentally different things?

In some situations, developers may want to inherit `OAuthFlow` to implement their own `OAuthFlow`. We already have such one: slack_bolt/adapter/aws_lambda/lambda_s3_oauth_flow.py So, I wanted to have the option to build the whole `OAuthFlow` outside the `App`/`AsyncApp`. If a developer just `OAuthFlow` as-is and wants to configure only `OAuthSettings`, the developer can go with `settings` and ignore `oauth_flow` here.

>2 Why is oauth_state_store available as an argument outside of oauth_settings?

I assume that just customizing two stores is a common pattern. In the case, the following is

```python
app = App(
    signing_secret=os.environ.get("SLACK_SIGNING_SECRET"),
    installation_store=FileInstallationStore(),
    state_store=FileOAuthStateStore(expiration_seconds=600),
)
```

a bit simpler than the one below.

```python
app = App(
    signing_secret=os.environ.get("SLACK_SIGNING_SECRET"),
    installation_store=FileInstallationStore(),
    oauth_settings=OAuthSettings(
        state_store=FileOAuthStateStore(expiration_seconds=600),
    )
)
```

That said, this one is a minor convenience and it just may cause some confusion. I'm fine to remove this. Thoughts?

Reply:
>2

We can add this later. I removed it in https://github.com/slackapi/bolt-python/pull/74/commits/89472694338032557521675cdb7dbb75bb2364f1